web-components/src/components/modal/CreditRetireModal.tsx

```diff
@@ -37,22 +13,15 @@ const CreditRetireModal: React.FC<CreditRetireModalProps> = ({
   batchDenom,
   open,
   onClose,
-}) => {
-  const styles = useStyles();
-
-  return (
-    <Modal className={styles.modal} open={open} onClose={onClose}>
-      <Title variant="h3" align="center" className={styles.mainTitle}>
-        Retire Credits
-      </Title>
-      <CreditRetireForm
-        holder={holder}
-        availableTradableAmount={1000}
-        batchDenom={batchDenom}
-        onClose={() => null}
-      />
-    </Modal>
-  );
-};
+}) => (
+  <FormModalTemplate title="Retire Credits" open={open} onClose={onClose}>
```

Comment:
@victordelval I've updated this to use `FormModalTemplate`, you might want to start from there when you start working on https://github.com/regen-network/regen-registry/issues/809